RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc

```diff
@@ -121,6 +127,48 @@ namespace {
     {
         return std::pair<bool,bool>(df.linkError(), df.capidError());
     }
+
+    std::unique_ptr<HBHEStatusBitSetter> parse_HBHEStatusBitSetter(
+        const edm::ParameterSet psdigi)
+    {
+        return std::make_unique<HBHEStatusBitSetter>(
+            psdigi.getParameter<double>("nominalPedestal"),
+            psdigi.getParameter<double>("hitEnergyMinimum"),
+            psdigi.getParameter<int>("hitMultiplicityThreshold"),
+            psdigi.getParameter<std::vector<edm::ParameterSet> >("pulseShapeParameterSets"));
+    }
+
+    std::unique_ptr<HBHEPulseShapeFlagSetter> parse_HBHEPulseShapeFlagSetter(
+        const edm::ParameterSet &psPulseShape)
+    {
+        return std::make_unique<HBHEPulseShapeFlagSetter>(
+            psPulseShape.getParameter<double>("MinimumChargeThreshold"),
+            psPulseShape.getParameter<double>("TS4TS5ChargeThreshold"),
```

Comment:
So, are you expecting instead that one would create a constructor-less struct and fill it out element by element, so that we don't have a function call with more than 5 arguments? Sorry, my 
imagination was unable to encompass that wisdom. I am afraid, I need a little bit more arguments explaining the advantages of this kind of design.


On 09/08/2016 08:46 AM, Slava Krutelyov wrote:
>
> In RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc <https://github.com/cms-sw/cmssw/pull/15779#discussion_r78008429>:
>
> > +    std::unique_ptr<HBHEStatusBitSetter> parse_HBHEStatusBitSetter(
> > +        const edm::ParameterSet psdigi)
> > +    {
> > +        return std::make_unique<HBHEStatusBitSetter>(
> > +            psdigi.getParameter<double>("nominalPedestal"),
> > +            psdigi.getParameter<double>("hitEnergyMinimum"),
> > +            psdigi.getParameter<int>("hitMultiplicityThreshold"),
> > +            psdigi.getParameter<std::vector<edm::ParameterSet> >("pulseShapeParameterSets"));
> > +    }
> > +
> > +    std::unique_ptr<HBHEPulseShapeFlagSetter> parse_HBHEPulseShapeFlagSetter(
> > +        const edm::ParameterSet &psPulseShape)
> > +    {
> > +        return std::make_unique<HBHEPulseShapeFlagSetter>(
> > +            psPulseShape.getParameter<double>("MinimumChargeThreshold"),
> > +            psPulseShape.getParameter<double>("TS4TS5ChargeThreshold"),
>
> Thank you for your educational comment about C++.
> the comment was about the constructor argument list.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15779/files/d201d93f931d7a63c81f8689979d4ffd6cdedbeb#r78008429>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-drpGj8H3zigntbyzXJvdgsrcsE8ks5qoBG4gaJpZM4J36UI>.
>